### ISSUE TITLE
[ARM32/Linux] Test failure: System.Numerics.Vectors.Tests.dll

### ISSUE THREAD

**Issue by hseok-oh** [dotnet/corefx#18601]
Environment: Raspberry pi 3 (Ubuntu 16.04)

```
System.Numerics.Tests.Vector2Tests.Vector2TransformNormalTest [FAIL]
   Assert.Equal() Failure
   Expected: <0.3169873, 2.183013>
   Actual:   <0.3169873, 2.183013>
   Stack Trace:
         at System.Numerics.Tests.Vector2Tests.Vector2TransformNormalTest()
System.Numerics.Tests.Vector2Tests.Vector2TransformNormal3x2Test [FAIL]
   Assert.Equal() Failure
   Expected: <-0.1339746, 2.232051>
   Actual:   <-0.1339746, 2.232051>
   Stack Trace:
         at System.Numerics.Tests.Vector2Tests.Vector2TransformNormal3x2Test()
```

Simple Test
```c#
public class Test
{
    const int Pass = 100;
    const int Fail = -1;
    const float Pi = (float)Math.PI;
    
    public static float ToRadians(float degrees)
    {
        return degrees * Pi / 180f;
    }
    
    public static int Main()
    {
        Vector2 v = new Vector2(1.0f, 2.0f);
        Matrix4x4 m =
            Matrix4x4.CreateRotationX(ToRadians(30.0f)) *
            Matrix4x4.CreateRotationY(ToRadians(30.0f)) *
            Matrix4x4.CreateRotationZ(ToRadians(30.0f));
        m.M41 = 10.0f;
        m.M42 = 20.0f;
        m.M43 = 30.0f;

        Vector2 expected = new Vector2(0.3169873f, 2.18301272f);
        Vector2 actual;

        actual = Vector2.TransformNormal(v, m);
        if(expected == actual)
        {
            Console.WriteLine("Pass");
            return Pass;
        }
        else
        {
            Console.WriteLine("Fail");
            return Fail;
        }
    }
}
```

Test result:
```
Fail
```

**Comment by hseok-oh** [comment]
It needs to use `MathHelper.Equal` method to compare `expected` and `actual` like other tests in `System.Numerics.Tests.Vector2Tests`.

**Comment by hseok-oh** [comment]
`Matrix4x4.CreateRotationX()` calls `MathF.Cos()`. And in MathF.Cos()` in CoreCLR uses cosf() function. But its result is different for different architecture. 

For example the result of, 
```
cosf(30.0f * 3.14159265358979323846f / 180.0f) 
```
ARM: 0x3f5db3d8
x64(x86): 0x3f5db3d7

The result is 1 bit difference.

**Comment by hseok-oh** [comment]
Fixed by #18658
